diff --git a/lib/coreBundle.js b/lib/coreBundle.js
index b2d70f2..fe95317 100644
--- a/lib/coreBundle.js
+++ b/lib/coreBundle.js
@@ -16652,8 +16652,7 @@ var init_crExecutionContext = __esm({
             ...handles.map((handle) => ({ objectId: handle._objectId }))
           ],
           returnByValue,
-          awaitPromise: true,
-          userGesture: true
+          awaitPromise: true
         }).catch(rewriteError);
         if (exceptionDetails)
           throw new JavaScriptErrorInEvaluate(getExceptionMessage(exceptionDetails));
@@ -36908,7 +36907,7 @@ var init_crNetworkManager = __esm({
         this._responseExtraInfoTracker.requestWillBeSentExtraInfo(event);
       }
       _onAuthRequired(sessionInfo, event) {
-        let response2 = "Default";
+        let response2 = "CancelAuth";
         const shouldProvideCredentials = this._shouldProvideCredentials(event.request.url);
         if (this._attemptedAuthentications.has(event.requestId)) {
           response2 = "CancelAuth";
